# Xen 4.8 prefers use of /dev/xen/privcmd instead of /proc/xen/privcmd
# so label it correctly and allow 'xl' access to it.
# Other Xen device node labels (blktap-2, xenbus_backend) are also corrected.
Index: refpolicy/policy/modules/kernel/devices.fc
===================================================================
--- refpolicy.orig/policy/modules/kernel/devices.fc
+++ refpolicy/policy/modules/kernel/devices.fc
@@ -186,6 +186,9 @@ ifdef(`distro_suse', `
 /dev/xen/gntdev		-c	gen_context(system_u:object_r:xen_device_t,s0)
 /dev/xen/gntalloc	-c	gen_context(system_u:object_r:xen_device_t,s0)
 /dev/xen/xenbus		-c	gen_context(system_u:object_r:xen_device_t,s0)
+/dev/xen/xenbus_backend	-c	gen_context(system_u:object_r:xen_device_t,s0)
+/dev/xen/privcmd		-c	gen_context(system_u:object_r:xen_device_t,s0)
+/dev/xen/hypercall		-c	gen_context(system_u:object_r:xen_device_t,s0)
 
 ifdef(`distro_debian',`
 # this is a static /dev dir "backup mount"
Index: refpolicy/policy/modules/contrib/xen.te
===================================================================
--- refpolicy.orig/policy/modules/contrib/xen.te
+++ refpolicy/policy/modules/contrib/xen.te
@@ -687,6 +687,7 @@ corenet_tcp_sendrecv_soundd_port(xm_t)
 dev_read_rand(xm_t)
 dev_read_urand(xm_t)
 dev_read_sysfs(xm_t)
+dev_rw_xen(xm_t)
 
 files_read_etc_runtime_files(xm_t)
 files_read_etc_files(xm_t)
